modules/ding_user_frontend/ding_user_frontend.module

```diff
@@ -330,3 +329,54 @@ function ding_user_frontend_build_menu($path = NULL) {
 
   return $menu;
 }
+
+/**
+ * Implements hook_outbound_alter().
+ *
+ * If URLs like user/1234/view are cached in e.g. varnish, then other users
+ * may be exposed to these URLs.
+ *
+ * This function normalizes the user/* paths, by replacing the uid with "me".
+ */
+function ding_user_frontend_url_outbound_alter(&$path, &$options, $original_path) {
+  global $user;
+  if (preg_match('/^user\/(\d+)\/(.*)$/', $path, $matches)) {
+    // As default change the path to use "me".
+    $path = 'user/me/' . $matches[2];
+
+    // To allow site administrators to edit other user they have to be filter
+    // out of this alter. When accessing another drupal user account.
+    if ($user->uid != $matches[1] && (array_search("administrators", $user->roles) || array_search("local administrator", $user->roles))) {
+      // This is an "administrators" or "local administrator" trying to edit
+      // another user. To "allow" this we reset the path.
+      $path = $original_path;
```

Comment:
The structure here seems odd:

1. L345 changes the path and L352 resets it. I would prefer if we check if all conditions are met and set it once. I appreciate the level of commenting explaining the situation though.
2. I think `array_search()` should be replaced with `in_array()`. We do not care about positioning and if one of the roles we are looking for have position 0 then that will evaluate to `false`.
3. I am considering whether we should do this check based on permissions instead of roles. Would that be possible?